src/views/virtualmachines/details/tabs/details/components/grid/leftGrid/VirtualMachineDetailsLeftGrid.tsx

```diff
@@ -75,6 +80,32 @@ const VirtualMachineDetailsLeftGrid: React.FC<VirtualMachineDetailsLeftGridProps
         <VirtualMachineDescriptionItem
           descriptionData={getAnnotation(vm, DESCRIPTION_ANNOTATION) || None}
           descriptionHeader={t('Description')}
+          isEdit
+          onEditClick={() => setDescriptionModalOpen(true)}
+        />
+        <DescriptionModal
+          obj={vm}
+          isOpen={descriptionModalOpen}
+          onClose={() => setDescriptionModalOpen(false)}
+          onSubmit={(updatedDescription) => {
```

Comment:
In my opinion, would be more readable if this function is extracted and moved above.